ecosystem/nft-metadata-crawler/src/asset_uploader/worker/mod.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    asset_uploader::worker::config::AssetUploaderWorkerConfig, config::Server,
+    utils::constants::MAX_ASSET_UPLOAD_RETRY_SECONDS,
+};
+use anyhow::Context;
+use axum::{
+    body::Body, http::StatusCode, response::IntoResponse, routing::post, Extension, Json, Router,
+};
+use reqwest::{multipart::Form, Client};
+use serde::Deserialize;
+use std::{sync::Arc, time::Duration};
+use tracing::{error, info};
+use url::Url;
+
+pub mod config;
+
+#[derive(Clone)]
+pub struct AssetUploaderWorkerContext {
+    config: Arc<AssetUploaderWorkerConfig>,
+}
+
+#[derive(Debug, Deserialize)]
+struct UploadRequest {
+    url: Url,
+}
+
+impl AssetUploaderWorkerContext {
+    pub fn new(config: AssetUploaderWorkerConfig) -> Self {
+        Self {
+            config: Arc::new(config),
+        }
+    }
+
+    /// Uploads an asset to Cloudflare and returns the response
+    async fn upload_asset(&self, url: &Url) -> anyhow::Result<impl IntoResponse> {
+        let hashed_url = sha256::digest(url.to_string());
+        let client = Client::builder()
+            .timeout(Duration::from_secs(MAX_ASSET_UPLOAD_RETRY_SECONDS))
+            .build()
+            .context("Error building reqwest client")?;
+        let form = Form::new()
+            .text("id", hashed_url.clone()) // Replace with actual metadata
+            .text("url", url.to_string());
+
+        info!(
+            asset_uri = ?url,
+            "[Asset Uploader] Uploading asset to Cloudflare"
+        );
+
+        let res = client
+            .post(format!(
+                "https://api.cloudflare.com/client/v4/accounts/{}/images/v1",
+                self.config.cloudflare_account_id
+            ))
+            .header(
+                "Authorization",
+                format!("Bearer {}", self.config.cloudflare_auth_key),
+            )
+            .multipart(form)
+            .send()
+            .await
+            .context("Error sending request to Cloudflare")?;
+
+        reqwest_response_to_axum_response(res).await
+    }
+
+    async fn handle_upload(
+        Extension(context): Extension<Arc<AssetUploaderWorkerContext>>,
+        Json(request): Json<UploadRequest>,
+    ) -> impl IntoResponse {
+        match context.upload_asset(&request.url).await {
+            Ok(res) => {
+                let res = res.into_response(); // TODO: How to log response body?
```

Comment:
wait which service is updating the db for request status and response? is it throttler or worker?

btw do we need to log the response if postgresdb will maintain request status and response?

Reply:
It's the throttler that performs the db updates, and it's just nice to log the body to help with debugging